### PR TITLE
Controllers Branch - Merging the old and new interfaces

### PR DESCRIPTION
**Pull request by jherico**
This merges the old scripting interface `ControllerScriptingInterface` defined in `interface` and derived from `AbstractControllerScriptingInterface` from `script-engine` with the new design... `controller::ScriptingInterface` defined in `controllers`.

* UserInputMapper and InputDevice have been moved to `Controllers`
* Dependency between `Controllers` and `InputPlugins` is reversed... `InputPlugins` now depends on `Controllers`.  Several superfluous dependencies from `InputPlugins` have been removed
*

### CODE REVIEW

File: libraries/controllers/src/controllers/ScriptingInterface.cpp [hunk 1]
```diff
@@ -347,6 +340,66 @@ namespace controller {
...
+    int ScriptingInterface::getNumberOfButtons() const {
+        return StandardButtonChannel::NUM_STANDARD_BUTTONS;
+    }
+
+    bool ScriptingInterface::isButtonPressed(int buttonIndex) const {
+        return getButtonValue((StandardButtonChannel)buttonIndex) == 0.0 ? false : true;
+    }
+
+    int ScriptingInterface::getNumberOfTriggers() const {
+        return 2;
```
Comment (ZappoMan) [inline libraries/controllers/src/controllers/ScriptingInterface.cpp]: magic

File: libraries/controllers/src/controllers/ScriptingInterface.cpp [hunk 2]
```diff
@@ -347,6 +340,66 @@ namespace controller {
...
+    }
+
+    float ScriptingInterface::getTriggerValue(int triggerIndex) const {
+        return getAxisValue(triggerIndex == 0 ? StandardAxisChannel::LT : StandardAxisChannel::RT);
+    }
+
+    int ScriptingInterface::getNumberOfJoysticks() const {
+        return 2;
```
Comment (ZappoMan) [inline libraries/controllers/src/controllers/ScriptingInterface.cpp]: magic

File: libraries/controllers/src/controllers/ScriptingInterface.cpp [hunk 3]
```diff
@@ -347,6 +340,66 @@ namespace controller {
...
+            yid = StandardAxisChannel::RY;
+        }
+        vec2 result;
+        result.x = getAxisValue(xid);
+        result.y = getAxisValue(yid);
+        return result;
+    }
+
+    int ScriptingInterface::getNumberOfSpatialControls() const {
+        return 2;
```
Comment (ZappoMan) [inline libraries/controllers/src/controllers/ScriptingInterface.cpp]: magic

File: libraries/input-plugins/src/input-plugins/InputPlugin.cpp [hunk 4]
```diff
@@ -22,7 +22,7 @@ InputPluginList getInputPlugins() {
     InputPlugin* PLUGIN_POOL[] = {
         new KeyboardMouseDevice(),
         new SDL2Manager(),
-        new SixenseManager(),
+        //new SixenseManager(),
```
Comment (ZappoMan) [inline libraries/input-plugins/src/input-plugins/InputPlugin.cpp]: please put this back

File: libraries/script-engine/src/ScriptEngine.cpp [hunk 5]
```diff
@@ -310,7 +309,7 @@ void ScriptEngine::init() {
 
     registerGlobalObject("Script", this);
     registerGlobalObject("Audio", &AudioScriptingInterface::getInstance());
-    registerGlobalObject("Controller", _controllerScriptingInterface);
+//    registerGlobalObject("Controller", _controllerScriptingInterface);
```
Comment (ZappoMan) [inline libraries/script-engine/src/ScriptEngine.cpp]: we still want this don't we???

File: interface/src/scripting/ControllerScriptingInterface.cpp [hunk 6]
```diff
@@ -395,96 +387,49 @@ glm::vec2 ControllerScriptingInterface::getViewportDimensions() const {
     return qApp->getUiSize();
 }
 
-QString ControllerScriptingInterface::sanatizeName(const QString& name) {
-    QString cleanName { name };
-    cleanName.remove(QRegularExpression{"[\\(\\)\\.\\s]"});
-    return cleanName;
-}
-
-void ControllerScriptingInterface::wireUpControllers(ScriptEngine* engine) {
```
Comment (ZappoMan) [inline interface/src/scripting/ControllerScriptingInterface.cpp]: where did this go?

**Comment by hifi-gustavo** [comment]
Build console output for commit: 0063f9ae1d9e10fce4ef4e25fe5e9cf477c42340 available at: https://gist.github.com/bec991cccc72ca19765d<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6080.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6080.dmg

File: libraries/input-plugins/src/input-plugins/InputPlugin.cpp (hunk 4 above)
Comment (jherico) [inline libraries/input-plugins/src/input-plugins/InputPlugin.cpp]: fixed

File: libraries/controllers/src/controllers/ScriptingInterface.cpp (hunk 1 above)
Comment (jherico) [inline libraries/controllers/src/controllers/ScriptingInterface.cpp]: fixed

File: libraries/controllers/src/controllers/ScriptingInterface.cpp (hunk 2 above)
Comment (jherico) [inline libraries/controllers/src/controllers/ScriptingInterface.cpp]: fixed

File: libraries/controllers/src/controllers/ScriptingInterface.cpp (hunk 3 above)
Comment (jherico) [inline libraries/controllers/src/controllers/ScriptingInterface.cpp]: fixed

File: libraries/script-engine/src/ScriptEngine.cpp (hunk 5 above)
Comment (jherico) [inline libraries/script-engine/src/ScriptEngine.cpp]: fixed

File: interface/src/scripting/ControllerScriptingInterface.cpp (hunk 6 above)
Comment (jherico) [inline interface/src/scripting/ControllerScriptingInterface.cpp]: Into the constructor of the base class.

**Comment by hifi-gustavo** [comment]
Build console output for commit: e39219c2b5a7e2d52b4de105d736c90015998255 available at: https://gist.github.com/bec991cccc72ca19765d<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6080.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6080.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 10df0f2d8c097889c32789a37ea2467eedd74e86 available at: https://gist.github.com/bec991cccc72ca19765d<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6080.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6080.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 195045a4ec8a29c11ef65a1cecf433bcca756c98 available at: https://gist.github.com/bec991cccc72ca19765d<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6080.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6080.dmg

File: libraries/controllers/src/controllers/ScriptingInterface.cpp [hunk 7]
```diff
@@ -116,12 +109,12 @@ namespace controller {
         return deviceMap;
     }
 
-    NewControllerScriptingInterface::NewControllerScriptingInterface() {
+    ScriptingInterface::ScriptingInterface() {
```
Comment (ZappoMan) [inline libraries/controllers/src/controllers/ScriptingInterface.cpp]: I think this is the source of the problem... basically, the input devices aren't available at the construction of the scripting interface and so at this point they don't exist in the input mapper.

**Pull request merged**